### DISCUSSION TITLE
Order of bound variables in binding

### DISCUSSION THREAD

**Discussion by hystricidae** [comunica/comunica#1344] [Q&A] [answered]
The order of the bound variables in a binding does not correspond to the order of the  variables as given in the SELECT query. E.g.: The order of variables in the live example is `?movie ?title ?name`. The results are however ordered as `?title ?name ?movie`. 
Is there a way to retain the original order of variables as given in the SPARQL query?

**Comment by rubensworks** [comment, marked as answer]
In SPARQL, bindings (or solution mappings) are just partial functions that map variables to RDF terms, but there is no defined order for variables specifically.
Are you experiencing a specific problem due to this?

**Reply by hystricidae** [reply]
Thanks for the reply and the clarification! The only minor "problem" is, that it was a bit confusing at first when the returned variables were "out of order" in regard to the order as within the query; it is however just a minor inconvenience.  I just expected an inherent order as seen f.e. with the Wikidata Query Service where the returned columns are shown in the same order as the variables in the select.

**Reply by rubensworks** [reply]
I assume you're using https://query.comunica.dev/. Feel free to open an issue or PR over at https://github.com/comunica/jQuery-Widget.js to modify the way bindings keys are ordered.

**Reply by balhoff** [reply]
I was also surprised by this (in my own application). It would be a nice convenience if `Bindings.keys` iterated in the same order as the SPARQL query. I'm not sure how this information would be propagated within Comunica though.

**Reply by rubensworks** [reply]
> It would be a nice convenience if Bindings.keys iterated in the same order as the SPARQL query. I'm not sure how this information would be propagated within Comunica though.

@balhoff That sounds reasonable indeed. Could you open a feature request issue, so we can add it to our backlog?
